src/pkg/aws/ecs/tail.go

```diff
@@ -24,17 +24,25 @@ func (a *AwsEcs) Tail(ctx context.Context, taskArn TaskArn) error {
 		return err
 	}
 
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	taskch := make(chan error)
+	defer close(taskch)
+	go func() {
+		taskch <- WaitForTask(ctx, taskArn, time.Second*3)
+	}()
+
 	spinMe := 0
 	for {
-		err = printLogEvents(ctx, es) // blocking
+		err = printLogEvents(ctx, es, taskch) // blocking
 		if err != nil {
 			return err
 		}
 
 		err := getTaskStatus(ctx, a.Region, a.ClusterName, taskId)
 		if err != nil {
 			// Before we exit, print any remaining logs (ignore errors)
-			printLogEvents(ctx, es)
+			printLogEvents(ctx, es, nil)
```

Comment:
this could block indefinitely if the task ends after the previous `printLogEvents` returns